bundles/org.openhab.binding.benqprojector/src/main/java/org/openhab/binding/benqprojector/internal/connector/BenqProjectorSerialConnector.java

```diff
@@ -0,0 +1,206 @@
+            }
+            this.in = null;
+        }
+        if (serialPort != null) {
+            logger.debug("Close serial port");
+            serialPort.close();
+            serialPort.removeEventListener();
+            this.serialPort = null;
+        }
+
+        logger.debug("Closed");
+    }
+
+    @Override
+    public String sendMessage(String data, int timeout) throws BenqProjectorException {
+        InputStream in = this.in;
+        OutputStream out = this.out;
+
+        if (in == null || out == null) {
+            connect();
+            in = this.in;
+            out = this.out;
+        }
+
+        try {
+            if (in != null && out != null) {
+                // flush input stream
+                if (in.markSupported()) {
+                    in.reset();
+                } else {
+                    while (in.available() > 0) {
+                        int availableBytes = in.available();
+
+                        if (availableBytes > 0) {
+                            byte[] tmpData = new byte[availableBytes];
+                            in.read(tmpData, 0, availableBytes);
+                        }
+                    }
+                }
+                return sendMmsg(data, timeout);
+            } else {
+                return BLANK;
+            }
+        } catch (IOException e) {
+            logger.debug("IO error occurred...reconnect and resend once: {}", e.getMessage());
+            disconnect();
+            connect();
+
+            try {
+                return sendMmsg(data, timeout);
+            } catch (IOException e1) {
+                throw new BenqProjectorException(e);
+            }
+        }
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent arg0) {
+    }
+
+    private String sendMmsg(String data, int timeout) throws IOException, BenqProjectorException {
+        String resp = BLANK;
+
+        InputStream in = this.in;
+        OutputStream out = this.out;
+
+        if (in != null && out != null) {
+            out.write((START + data + END).getBytes(StandardCharsets.US_ASCII));
+            out.flush();
+
+            long startTime = System.currentTimeMillis();
+            long elapsedTime = 0;
+
+            while (elapsedTime < timeout) {
+                int availableBytes = in.available();
+                if (availableBytes > 0) {
+                    byte[] tmpData = new byte[availableBytes];
+                    int readBytes = in.read(tmpData, 0, availableBytes);
+                    resp = resp.concat(new String(tmpData, 0, readBytes, StandardCharsets.US_ASCII));
+                    if (resp.contains(END)) {
+                        return resp.replaceAll("[\\r\\n*#>]", BLANK).replace(data, BLANK);
+                    }
+                } else {
+                    try {
+                        Thread.sleep(100);
+                    } catch (InterruptedException e) {
+                        throw new BenqProjectorException(e);
```

Comment:
I removed the ability to specify the timeout since it was not needed for this binding. The hard-coded timeout is 5000 ms. I am not sure how it could be 130s.

Reply:
POWER_OFF_TIMEOUT is 130 * 1000.

Reply:
Still confused... POWER_OFF_TIMEOUT was not done here, it is in epson: https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/EpsonProjectorDevice.java

Reply:
Indeed, I imported the wrong project. The sleep is a bit ugly, but if I understand your code correctly, the communication should be finished way quicker than the timeout under normal circumstances.

Reply:
Yes the response should be received by the time that the first 100ms sleep has elapsed. The loop will keep checking for a response after each sleep until 5 seconds have elapsed.